tests/AttributeTest.php

```diff
@@ -19,13 +21,43 @@
 use Tnapf\JsonMapper\Attributes\StringType;
 use Tnapf\JsonMapper\Exception\InvalidArgumentException;
 use Tnapf\JsonMapper\Exception\InvalidValueTypeException;
+use Tnapf\JsonMapper\MapperInterface;
 use Tnapf\JsonMapper\Tests\Fakes\IssueCategory;
 use Tnapf\JsonMapper\Tests\Fakes\IssueState;
 use Tnapf\JsonMapper\Tests\Fakes\RolePermission;
 use Tnapf\JsonMapper\Tests\Fakes\AttributeDuplication;
 
 class AttributeTest extends TestCase
 {
+    public function testCallbackType(): void
+    {
+        $callbackType = new class ('test', true) extends CallbackType {
+            public function isType(mixed $data): bool
+            {
+                return is_int($data);
+            }
+
+            public function map(mixed $data, MapperInterface $mapper): mixed
+            {
+                return $data + 1;
```

Comment:
Looks like a misuse of a type. This method implements a specific logic that's beyond the scope of "converting" data.

Reply:
Not sure I understand

Reply:
> Not sure I understand

Well, `+ 1` usually is not something that transforms one type of data into another (it only does as a side effect).

Mapping string to integer or array to object doesn't actually change the data itself (excluding the cases of string `'asdf'` to int `0`).

I understand that it doesn't really matter for that specific test, but I prefer seeing tests use features as intended. I consider modifying the input data to be an unexpected behavior of a mapper.

Reply:
No I see, it makes sense. I can update the test to make it more accurate